packages/playground/cases/react/basic/src/App.jsx

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import "./index.css";
+import { ContextComponent } from "./CountProvider";
+import { ReactRefreshFinder } from "./ReactRefreshFinder";
+
+const Button = () => {
+	const [count, setCount] = React.useState(10);
+	return <button onClick={() => setCount(count => count + 1)}>{count}</button>;
+};
+
+export const App = () => {
+	return (
+		<div className="App">
+			<div className="header">Hello World</div>
+			<Button />
+			<div className="placeholder">__EDITED__</div>
```

Comment:
Is this correct? I don't know if its good choice to directly consume the source file for E2E testing or do we have temporary directories for testing?

Reply:
All e2e tests are run in a temporary directory and the `pathInfo` fixture will copy them.

Reply:
But is this supposed to be this?
```suggestion
			<div className="placeholder"> __PLACE_HOLDER__ </div>
```

Reply:
Yes, I fixed it and added a line to avoid it

https://github.com/web-infra-dev/rspack/blob/b0f95ae2c5f8ba4b760fa99957de471527f229db/packages/playground/cases/react/basic/index.test.ts#L11